src/transformers/generation/utils.py

```diff
@@ -2352,6 +2354,22 @@ def generate(
         else:
             input_ids = inputs_tensor if model_input_name == "input_ids" else model_kwargs.pop("input_ids")
 
+        # Expand inputs depending on the generation mode
+        if generation_config.num_beams > 1:
+            input_ids, model_kwargs = self._expand_inputs_for_generation(
+                input_ids=input_ids,
+                expand_size=generation_config.num_beams,
+                is_encoder_decoder=self.config.is_encoder_decoder,
+                **model_kwargs,
+            )
+        elif generation_config.do_sample:
+            input_ids, model_kwargs = self._expand_inputs_for_generation(
+                input_ids=input_ids,
+                expand_size=generation_config.num_return_sequences,
+                is_encoder_decoder=self.config.is_encoder_decoder,
+                **model_kwargs,
+            )
```

Comment:
```suggestion
        input_ids, model_kwargs = self._expand_inputs_for_generation(
            input_ids=input_ids,
            expand_size=max(generation_config.num_beams, generation_config.num_return_sequences)
            is_encoder_decoder=self.config.is_encoder_decoder,
            **model_kwargs,
        )
```

we can take advantage of:
- `num_beams` and `num_return_sequences` both defaulting to `1`
- `expand_size=1` is a no-op

to make it shorter :D